src/passes/OptimizeAddedConstants.cpp

```diff
@@ -222,7 +242,7 @@ template<typename P, typename T> class MemoryAccessOptimizer {
 
   // Sees if we can optimize a particular constant.
   Result canOptimizeConstant(Literal literal) {
-    auto value = literal.geti32();
+    uint64_t value = memory64 ? literal.geti64() : literal.geti32();
```

Comment:
```suggestion
    uint64_t value = literal.getInteger();
```
There is a helper for this pattern.

Reply:
address comment, thanks